cow_instrument/DetectorInfo.h

```diff
@@ -181,13 +229,72 @@ DetectorInfo<InstTree>::DetectorInfo(std::shared_ptr<InstTree> &&instrumentTree)
       m_positions(std::make_shared<std::vector<Eigen::Vector3d>>(m_nDetectors)),
       m_rotations(
           std::make_shared<std::vector<Eigen::Quaterniond>>(m_nDetectors)),
+      m_linearIndexMap(makeDefaultIndexes(instrumentTree)),
+      m_durations(std::make_shared<const ScanTimes>(1, scanTime)),
       m_pathComponentInfo(
           std::forward<std::shared_ptr<InstTree>>(instrumentTree)) {
 
   init();
 }
 
 template <typename InstTree>
+template <typename InstSptrType, typename TimeIndexesType,
+          typename ScanTimesType, typename PositionsType,
+          typename RotationsType>
+DetectorInfo<InstTree>::DetectorInfo(InstSptrType &&instrumentTree,
+                                     TimeIndexesType &&timeIndexes,
+                                     ScanTimesType &&scanTimes,
+                                     PositionsType &&positions,
+                                     RotationsType &&rotations)
+    : m_l2Paths(SourceSampleDetectorPathFactory<InstTree>{}.createL2(
+          *instrumentTree)),
+      m_l1Paths(SourceSampleDetectorPathFactory<InstTree>{}.createL1(
+          *instrumentTree)),
+      m_nDetectors(instrumentTree->nDetectors()),
+      m_l1(std::make_shared<L1s>(m_nDetectors)),
+      m_l2(std::make_shared<L2s>(m_nDetectors * scanTimes.size())),
+      m_isMasked(std::make_shared<MaskFlags>(m_nDetectors, Bool(false))),
+      m_isMonitor(std::make_shared<MonitorFlags>(m_nDetectors, Bool(false))),
+      m_detectorComponentIndexes(std::make_shared<const std::vector<size_t>>(
+          instrumentTree->detectorComponentIndexes())),
+      m_positions(std::make_shared<std::vector<Eigen::Vector3d>>(
+          std::forward<PositionsType>(positions))),
+      m_rotations(std::make_shared<std::vector<Eigen::Quaterniond>>(
+          std::forward<RotationsType>(rotations))),
+      m_linearIndexMap(std::make_shared<std::vector<std::vector<size_t>>>(
+          std::forward<TimeIndexesType>(timeIndexes))),
+      m_durations(
+          std::make_shared<ScanTimes>(std::forward<ScanTimesType>(scanTimes))),
+      m_pathComponentInfo(
+          std::forward<std::shared_ptr<InstTree>>(instrumentTree)),
+      m_isScanning(true) {
+  size_t indexEntryCount = 0;
+  for (auto &positionTimes : *m_linearIndexMap) {
+    // Count up the time indexes for each component
+    indexEntryCount += positionTimes.size();
+  }
+  if (indexEntryCount != m_positions->size()) {
```

Comment:
@OwenArnold Maybe it would then be easier to just have a independent scan times (and number of scan times) for each detector.
It think it is enough to figure out those details once we implement it, no need for a more detailed prototype, I think?